Compile failed with build.sh and clang 6.0

It seems that clang treat all enum variable as unsigned, so compare unsigned to 0 should always be >= 

```
[  0%] Building CXX object pal/src/CMakeFiles/Chakra.Pal.Singular.dir/shmemory/shmemory.cpp.o
/home/xm1994/ChakraCore/pal/src/shmemory/shmemory.cpp:904:16: error: comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-unsigned-enum-zero-compare]
    if(element < 0 || element >= SIID_LAST)
       ~~~~~~~ ^ ~
/home/xm1994/ChakraCore/pal/src/shmemory/shmemory.cpp:946:16: error: comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-unsigned-enum-zero-compare]
    if(element < 0 || element >= SIID_LAST)
       ~~~~~~~ ^ ~
2 errors generated.
```

--- a/pal/src/shmemory/shmemory.cpp
+++ b/pal/src/shmemory/shmemory.cpp
@@ -901,7 +901,7 @@ SHMPTR SHMGetInfo(SHM_INFO_ID element)
     SHM_FIRST_HEADER *header = NULL;
     SHMPTR retval = 0;
 
-    if(element < 0 || element >= SIID_LAST)
+    if(element >= SIID_LAST)
     {
         ASSERT("Invalid SHM info element %d\n", element);
         return 0;
@@ -943,7 +943,7 @@ BOOL SHMSetInfo(SHM_INFO_ID element, SHMPTR value)
 {
     SHM_FIRST_HEADER *header;
 
-    if(element < 0 || element >= SIID_LAST)
+    if(element >= SIID_LAST)
     {
         ASSERT("Invalid SHM info element %d\n", element);
         return FALSE;

